oak/server/storage/storage_processor.cc

```diff
@@ -0,0 +1,211 @@
+  read_request.set_datum_name(name_or.ValueOrDie());
+
+  grpc::ClientContext context;
+  StorageReadResponse read_response;
+  grpc::Status status = storage_service_->Read(&context, read_request, &read_response);
+  // TODO: Propagate error status.
+  if (status.ok()) {
+    asylo::StatusOr<std::string> value_or =
+        DecryptDatum(read_response.datum_value(), DatumType::VALUE);
+    *datum_value = value_or.ValueOrDie();
+  }
+}
+
+void StorageProcessor::Write(const std::string& storage_name, const std::string& datum_name,
+                             const std::string& datum_value, const std::string& transaction_id) {
+  StorageWriteRequest write_request;
+  write_request.set_storage_id(GetStorageId(storage_name));
+  if (!transaction_id.empty()) {
+    write_request.set_transaction_id(transaction_id);
+  }
+  // TODO: Propagate error status.
+  asylo::StatusOr<std::string> name_or = EncryptDatum(datum_name, DatumType::NAME);
+  write_request.set_datum_name(name_or.ValueOrDie());
+
+  // TODO: Propagate error status.
+  asylo::StatusOr<std::string> value_or = EncryptDatum(datum_value, DatumType::VALUE);
+  write_request.set_datum_value(value_or.ValueOrDie());
+
+  grpc::ClientContext context;
+  StorageWriteResponse write_response;
+  grpc::Status status = storage_service_->Write(&context, write_request, &write_response);
+}
+
+void StorageProcessor::Delete(const std::string& storage_name, const std::string& datum_name,
+                              const std::string& transaction_id) {
+  StorageDeleteRequest delete_request;
+  delete_request.set_storage_id(GetStorageId(storage_name));
+  if (!transaction_id.empty()) {
+    delete_request.set_transaction_id(transaction_id);
+  }
+  // TODO: Propagate error status.
+  asylo::StatusOr<std::string> name_or = EncryptDatum(datum_name, DatumType::NAME);
+  delete_request.set_datum_name(name_or.ValueOrDie());
+
+  grpc::ClientContext context;
+  StorageDeleteResponse delete_response;
+  grpc::Status status = storage_service_->Delete(&context, delete_request, &delete_response);
+}
+
+void StorageProcessor::Begin(const std::string& storage_name, std::string* transaction_id) {
+  CHECK(transaction_id != nullptr);
+
+  StorageBeginRequest begin_request;
+  begin_request.set_storage_id(GetStorageId(storage_name));
+
+  grpc::ClientContext context;
+  StorageBeginResponse begin_response;
+  grpc::Status status = storage_service_->Begin(&context, begin_request, &begin_response);
+  if (status.ok()) {
+    *transaction_id = begin_response.transaction_id();
+  }
+}
+
+void StorageProcessor::Commit(const std::string& storage_name, const std::string& transaction_id) {
+  StorageCommitRequest commit_request;
+  commit_request.set_storage_id(GetStorageId(storage_name));
+  commit_request.set_transaction_id(transaction_id);
+
+  grpc::ClientContext context;
+  StorageCommitResponse commit_response;
+  grpc::Status status = storage_service_->Commit(&context, commit_request, &commit_response);
```

Comment:
Check status, or add TODO? (and elsewhere)

Reply:
Done.